uber/site_sections/registration.py

```diff
@@ -804,12 +804,18 @@ def review(self, session):
     def discount(self, session, message='', **params):
         attendee = session.attendee(params)
         if 'first_name' in params:
+            if 'overridden_price' in params:
+                if isinstance(attendee.overridden_price, str):
```

Comment:
I like the typecasting. It is much simpler than my solution. I've
substituted it and will be committing momentarily.

On Fri, Jul 22, 2016 at 4:31 PM, Victoria Earl <notifications@github.com>
wrote:

> In uber/site_sections/registration.py
> <https://github.com/magfest/ubersystem/pull/1903#discussion_r71959429>:
>
> > @@ -804,12 +804,18 @@ def review(self, session):
> >      def discount(self, session, message='', **params):
> >          attendee = session.attendee(params)
> >          if 'first_name' in params:
> > +            if 'overridden_price' in params:
> > +                if isinstance(attendee.overridden_price, str):
>
> HTML forms always pass a string - did you test this to make sure that
> isinstance is actually returning true as long as overridden_price can be
> typecast to an integer? The Python docs don't really indicate that the
> function would do that, but I'm not familiar enough to be sure (and it may
> be being typecast to int elsewhere, like when you load the session).
>
> Alternatively, you can try to typecast to an integer and use a try/except
> block instead. So like:
>
> try:
> attendee.overridden_price = int(attendee.overridden_price)
> except ValueError:
> attendee.overridden_price = -1
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/magfest/ubersystem/pull/1903/files/800896cca843f09a11c1a10cad930788cf46cb00#r71959429>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGPGAe-0B1LkA8Tl728eE4FS5w-SaWbbks5qYVLKgaJpZM4JTLyq>
> .
>